mypyc/test-data/run-strings.test

```diff
@@ -639,3 +639,67 @@ def test_encode() -> None:
     assert u'\u00E1'.encode('latin1') == b'\xe1'
     with assertRaises(UnicodeEncodeError):
         u.encode('latin1')
+
+[case testCompare]
+def test_compare() -> None:
+    s1: str = 'xxx'
+    s2: str = 'xxx'
+    s3: str = 'yyy'
+
+    assert s1 == s2
+    assert s2 == s1
+    assert s1 != s3
+    assert s3 != s1
+    assert s2 != s3
+    assert s3 != s2
+
+    assert s1 == s1
+    assert s2 == s2
+    assert s3 == s3
```

Comment:
A lot of other tests would break if basic string equality checks were busted (including the `testStrBasics` case at the top of this file), this is redundant :)